src/main/io/gps.c

```diff
@@ -1414,7 +1457,7 @@ void gpsUpdate(timeUs_t currentTimeUs)
 
         case GPS_STATE_RECEIVING_DATA:
 #ifdef USE_GPS_UBLOX
-            if (gpsConfig()->provider != GPS_MSP) {
+            if (gpsConfig()->provider != GPS_MSP && gpsConfig()->provider != GPS_VIRTUAL) {
```

Comment:
IMO this should be inverted now:
```suggestion
            if (gpsConfig()->provider == GPS_UBLOX || gpsConfig()->provider == GPS_NMEA) {
```

Also, it should probably test only for GPS_UBLOX; it seems strange to send ublox message to nmea GPS. But do not change behavior in this PR, just add `// TODO` to handle it later.